implementation-details.md

```diff
@@ -284,32 +313,53 @@ The `CrdsFilter`s are partitioned in pull requests between peers selected random
 
   where: 
   * `node_stake` - stake of "our" node, 
-  * `peer_stake[i]` - stake of i-th node from the list.
+  * `peer_stake[i]` - stake of i-th peer from the list.
+
   
-The higher the node's weight, the more pull requests will be associated with it.
+The higher the node's weight, the more pull requests will be sent to it.
 
-Each pull request contains one of the `CrdsFilter`s from the list and a `CrdsValue`, which is usually a `LegacyContactInfo`. It contains details of the node that sent the pull request - a list of its IP addresses, public key and wallclock. 
+Each pull request contains one of the `CrdsFilter`s from the list and a `CrdsValue`, which is a `LegacyContactInfo`. It contains details of the node that sent the pull request - a list of its IP addresses, public key and wallclock. 
 
-When a node receives a pull request it inserts the pull request value into its `crds`, or updates the existing one in case of duplicate. Then, it checks whether the pull request comes from a valid address as a pull response needs to be sent back. Next, the node checks the pull request wallclock value - if it's not within 15 seconds of the current node's wallclock the request is ignored. Otherwise, a node gathers data from the `crds` table and filters it with the provided `CrdsFilter`. Node uses the `mask_bits` value and the `CrdsShards` structure to find the proper hashes of `CrdsValue`s from `crds`:
-1. If `mask_bits = shard_bits = 12` the `CrdsFilter` `mask` value can be used directly as an index in the `shards` vector. The node will gather all `CrdsValue`s from `crds` whose hashes belong to that index in `shards` and filter them with the provided bloom filter. Values not existing in the filter will be collected for further processing.
-> _Example_
->
-> If `mask = 000000010000 = 16` we search for hashes from `shards` vector under index 16
+When a node receives a pull request it inserts the pull request value, `LegacyContactInfo`, into its `crds`, or updates the existing one in case of duplicate. Then, it checks whether the pull request comes from a valid address as a pull response needs to be sent back. Next, the node checks the pull request wallclock value - if it's not within 15 seconds of the current node's wallclock the request is ignored. Otherwise, a node gathers data from the `crds` table and filters it with the provided `CrdsFilter`. 
```

Comment:
> Otherwise, a node gathers data from the `crds` table and filters it with the provided `CrdsFilter`. 

"Otherwise, a node gathers its own data from the `crds` table and checks if the data it holds exists in the provided `CrdsFilter`."